src/client/components/sms-fraud/PhoneNumberForm.tsx

```diff
@@ -0,0 +1,65 @@
+import { FunctionComponent } from 'react';
+import { TEST_IDS } from '../../testIDs';
+import { SendMessageButton } from './SendSMSMessageButton';
+import classNames from 'classnames';
+import styles from './smsVerificationFraud.module.scss';
+import formStyles from '../../../styles/forms.module.scss';
+import { SendMessageMutation } from '../../../pages/sms-fraud';
+
+type PhoneNumberFormProps = {
+  phoneNumber: string;
+  setPhoneNumber: (phoneNumber: string) => void;
+  sendMessageMutation: SendMessageMutation;
+  email: string;
+  setEmail: (email: string) => void;
+};
+
+export const PhoneNumberForm: FunctionComponent<PhoneNumberFormProps> = ({
+  phoneNumber,
+  setPhoneNumber,
+  email,
+  setEmail,
+  sendMessageMutation,
```

Comment:
If we want to pass the mutation as prop here, I think it would be better to abstract it a bit more into following props: `onSubmit` and `isLoading`. 
Otherwise we might as well just call the mutation hook here, rather than passing it as a prop.

Reply:
> If we want to pass the mutation as prop here, I think it would be better to abstract it a bit more into following props: onSubmit and isLoading.

hmm I am passing the whole mutation because I am passing it all down to `SendMessageButton`. It needs the whole mutation, its using `mutate`, `data`, `error`, `isLoading`. 

* `SendMessageButton` is a separate component because it's present twice in both form steps.
* I need these two buttons to share the mutation state.

> Otherwise we might as well just call the mutation hook here, rather than passing it as a prop.

Unlike queries, [mutations do not share state](https://github.com/TanStack/query/issues/2304), each `mutate` call creates a separate mutation state.

The simplest way to achieve my desired shared state outcome is therefore to pull the mutation definition up the component tree and pass it down where needed. (a bit of prop drilling for sure, but seems simpler than the alternatives) 

Does it make sense? Am I missing something?

Reply:
Okay, makes sense to me.

But, consider if we are using all of the properties from this mutation. We probably do not, in that case it would be more readable to only pass the props that we actually need:
- `mutation.mutate` -> `onSubmit`
- `mutation.isLoading` -> `isLoading`
...etc.

WDYT?

Reply:
> But, consider if we are using all of the properties from this mutation.

I think it's a mixed bag :/ I think if we were passing just 1-2 properties, and passing them just one level down, I would agree it's more readable to pass them individually. 

But we are passing seemingly all of the main properties that you need to make full use of the mutation under normal circumstances (mutate, loading, error, data). I intend that the child component has full access to the mutation, so the code matches that. 

Also we are passing it 2 levels down, for the reasons stated above. Passing props individually in this situation seems to have these disadvantages: 

- It's more code than before (and it's repetitive)
- It forces us to manually type the individual props instead of inferring the types from ReactQuery, potentially introducing type errors/inaccuracies.
- Any time we change some of the types or we want to use another property from the mutation, we now have to make changes in 3 files instead of 1.

"Only pass down the things you need" is usually a good heuristic, but in this case, it seems passing down the entire mutation has more advantages. 

![image](https://github.com/fingerprintjs/fingerprintjs-pro-use-cases/assets/11249993/25665b39-a0a9-4816-84de-62a39d3d7145)
![image](https://github.com/fingerprintjs/fingerprintjs-pro-use-cases/assets/11249993/68873ee7-3d07-4628-9957-9cd6f4f39156)
![image](https://github.com/fingerprintjs/fingerprintjs-pro-use-cases/assets/11249993/94f569f3-6fd1-4ffc-8927-070ba3d496ee)